tests/system/Autoloader/AutoloaderTest.php

```diff
@@ -50,13 +56,15 @@ protected function setUp(): void
 
         $this->loader = new Autoloader();
         $this->loader->initialize($config, $modules)->register();
+
+        $this->classLoader = $this->getPrivateMethodInvoker($this->loader, 'loadInNamespace');
```

Comment:
Why do you need to test protected `loadInNamespace()`?

Reply:
Out of the scope of this PR, but would the following code be necessary?
It seems not, and the `loadInNamespace()` method doesn't seem necessary either.

https://github.com/codeigniter4/CodeIgniter4/blob/31b79b163528c7ee0f27ebad52f13c890186a994/system/Autoloader/Autoloader.php#L263-L264

Reply:
I think we don't need this test either:
https://github.com/codeigniter4/CodeIgniter4/blob/31b79b163528c7ee0f27ebad52f13c890186a994/tests/system/Autoloader/AutoloaderTest.php#L149-L156

PHP autoloader never passes the value `'\App\Controllers\Home.php'`.

Reply:
> Out of the scope of this PR, but would the following code be necessary?
> It seems not, and the `loadInNamespace()` method doesn't seem necessary either.

Those are necessary as the callback for `spl_autoload_register` does not expect leading `\`. For the trimming of `.php` part I think not. I think it's a misuse that a classname ends with .php

Reply:
> I think we don't need this test either:
> 
> https://github.com/codeigniter4/CodeIgniter4/blob/31b79b163528c7ee0f27ebad52f13c890186a994/tests/system/Autoloader/AutoloaderTest.php#L149-L156
> 
> PHP autoloader never passes the value `'\App\Controllers\Home.php'`.

This test is to checks if the value passed to `loadInNamespace` is valid (i.e., a class with no leading `\` and no `.php` part). So basically this is what the previous test is.

Reply:
What do you mean?
Is there a way to pass the classname with leading `\`?
https://3v4l.org/THIOb

Reply:
Okay, I think I misunderstood the manual. In that case, the removal of leading `\` seems unnecessary. How about the removal of `.php` extension ?

Reply:
We don't need the removal of `.php` extension. 
It is not a valid classname, so the method never receive such string.